lms/djangoapps/courseware/views/views.py

```diff
@@ -354,13 +354,16 @@ def course_info(request, course_id):
             if request.GET.get('upgrade', 'false') == 'true':
                 store_upgrade_cookie = True
                 show_upgrade_notification = True
-            elif upgrade_cookie_name in request.COOKIES and bool(request.COOKIES[upgrade_cookie_name]):
+            elif upgrade_cookie_name in request.COOKIES and course_id in request.COOKIES[upgrade_cookie_name]:
                 show_upgrade_notification = True
 
             if show_upgrade_notification:
                 upgrade_data = VerifiedUpgradeDeadlineDate(course, user)
                 if upgrade_data.is_enabled:
                     upgrade_link = upgrade_data.link
+                else:
```

Comment:
Depends on the desired outcome as we discussed.  I was avoiding adding a cookie when it was not needed.  I can change it if needed, but I do not see the need to add the cookie when there is no need.